### PR TITLE
[#492] Rewrite timebar to use a fixed-width <canvas>

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

File: src/timeline/timebar.js [hunk 1]
```diff
@@ -14,45 +14,50 @@ define( [ "util/lang", "./scrubber" ], function( util, Scrubber ) {
...
 
       _canvasContainer.style.width = containerWidth + "px";
 
-      _canvas.style.width = width + "px";
-
       var context = _canvas.getContext( "2d" );
 
       _canvas.height = _canvas.offsetHeight;
-      _canvas.width = _canvas.offsetWidth;
+      _canvas.width = containerWidth;
```
Comment (secretrobotron) [inline src/timeline/timebar.js]: Can we move this out of here, actually? My fear is that silly things like this still happen: http://stackoverflow.com/questions/2142535/how-to-clear-the-canvas-for-redrawing

File: src/timeline/timebar.js [hunk 2]
```diff
@@ -14,45 +14,50 @@ define( [ "util/lang", "./scrubber" ], function( util, Scrubber ) {
...
           padding = 20,
           lastPosition = 0,
-          lastTimeDisplayed = -( ( textWidth + padding ) / 2 );
+          lastTimeDisplayed = -( ( textWidth + padding ) / 2 ),
+          start = _tracksContainer.element.scrollLeft / inc,
+          end = ( _tracksContainer.element.scrollLeft + containerWidth ) / inc;
 
       context.clearRect ( 0, 0, _canvas.width, _canvas.height );
-
+      context.translate( -_tracksContainer.element.scrollLeft, 0 );
```
Comment (secretrobotron) [inline src/timeline/timebar.js:51]: So, if we're just drawing inside the width of the viewport now, why are translations by the value of `scrollLeft` necessary?

File: src/timeline/timebar.js [hunk 3]
```diff
@@ -93,16 +98,24 @@ define( [ "util/lang", "./scrubber" ], function( util, Scrubber ) {
       // stroke color
       context.strokeStyle = "#999999";
       context.stroke();
-      context.closePath();
+      context.translate( _tracksContainer.element.scrollLeft, 0 );
 
       _scrubber.update( containerWidth, zoom );
```
Comment (secretrobotron) [inline src/timeline/timebar.js:106]: You can completely skip passing `zoom` into this function now if you just call `scrubber.update` in the event listener and `update` functions below.

File: src/timeline/timebar.js (hunk 1 above)
Comment (jbuck) [inline src/timeline/timebar.js]: I'm setting the width to match the container width, which can be less than the width of the screen with a small video. I clear the canvas on line 47 below

Comment (secretrobotron) [inline src/timeline/timebar.js]: Right, and that's accurate. But, it looks like it is (or recently was) used as a way to clear the canvas as well. So, in effect, you'd be clearing it twice. Can we set the width less often than every scroll update?

File: src/timeline/timebar.js (hunk 2 above)
Comment (jbuck) [inline src/timeline/timebar.js:51]: scott's drawing algorithm assumes that it's drawing the entire timebar in one pass (look above, the inc is for the scrollWidth, not the canvas' width anymore). I translate by -scrollLeft to bring the part that should be drawn into the drawable part of the canvas. Getting rid of the translate calls here would require rewriting the drawing algorithm below, which is why I stuck with just translating the canvas and... manual culling haha

File: src/timeline/timebar.js (hunk 3 above)
Comment (jbuck) [inline src/timeline/timebar.js:106]: If I move the scrubber.update calls into the event listeners, then that means I need to duplicate the containerWidth calculations into both functions. That doesn't sound like a win to me.

I will remove the _lastZoom member though, as the scrubber already stores the previous zoom level

File: src/timeline/timebar.js (hunk 2 above)
Comment (secretrobotron) [inline src/timeline/timebar.js:51]: Ok, I think I get it now. Filing a "profiling" ticket.

Comment (jbuck) [inline src/timeline/timebar.js:51]: Covered by https://webmademovies.lighthouseapp.com/projects/65733/tickets/523-decouple-timeline-from-dom-widthheight maybe?

File: src/timeline/timebar.js (hunk 3 above)
Comment (secretrobotron) [inline src/timeline/timebar.js:106]: I didn't even see the containerWidth calc. Nvm. This is fine :).

**Pull request merged**